File: Basics_with_Python/PythonBasicCourse/4_For_Loop/10_For_Loop___Exercise/04._Clever_Lily.py
n = int(input())
washer_price = float(input())
toy_price = float(input())
toy_count = 0
money_presents = 0
addition = 10
for number in range(1, n + 1):
    if number % 2 == 0:
        money_presents += addition - 1
        addition += 10
    else:
        toy_count += 1
toy_money = toy_count * toy_price
saves = toy_money + money_presents
if saves >= washer_price:
    print(f"Yes! {abs(washer_price - saves):.2f}")
else:
    print(f"No! {abs(washer_price - saves):.2f}")

File: Basics_with_JavaScript/910_For_Loop/10_For_Loop_Excercise/04._Clever_Lily.js
function vladi(input) {
  let ages = Number(input[0]);
  let washMPrice = Number(input[1]);
  let toyPrice = Number(input[2]);
  let sumMoneyPresents = 0;
  let toyMoneyPresents = 0;
  let moneyPresentAmount = 0;
  let diff = 0;

  for (let age = 1; age <= ages; age++) {
    if (age % 2 === 0) {
      moneyPresentAmount += 10;
      sumMoneyPresents += moneyPresentAmount - 1;
    } else {
      toyMoneyPresents += toyPrice;
    }
  }

  diff = (Math.abs(washMPrice - (toyMoneyPresents + sumMoneyPresents))).toFixed(2);
  if (washMPrice <= toyMoneyPresents + sumMoneyPresents) {
    console.log(`Yes! ${diff}`);
  } else {
    console.log(`No! ${diff}`);
  }
}

vladi(["10", "170.00", "6"]);
vladi(["21", "1570.98", "3"]);
